### PR TITLE
Fetch events live from Eventful API

### PR DESCRIPTION
**Pull request by liuche**
I ended up in some kind of horrible circular dependency trap, and I didn't want to do a refactor of request_handler along with these changes. I'll file an issue to clean up request_handler though, mainly separating out the database handling, the location calls, and the event calls.

Basically, these all end up being kind of dependent because the request_handler contains all the db writing code, and imported the events.py code for event handling, but the events.py code also handles fetching venues for events (from Pull #45).

This is rebased on top of Pull #45 .

### CODE REVIEW

File: app/request_handler.py [hunk 1]
```diff
@@ -141,11 +164,90 @@ def searchLocation(lat, lng):
...
+    geo   = representation._geoRecord(float(eventObj["coordinates"]["lat"]), float(eventObj["coordinates"]["lng"]))
 
     db.child(eventsTable).update(
       {
         "details/" + key: event,
         "locations/" + key: geo
       }
     )
+
+def getEventfulEventObj(event):
```
Comment (jhugman) [inline app/request_handler.py:212]: Definitely getting to the limit of size/responsibility for `request_handler.py`. We're putting down tech debt here to have everything in this file.

File: app/request_handler.py [hunk 2]
```diff
@@ -141,11 +164,90 @@ def searchLocation(lat, lng):
...
+    yelpBiz = search._guessYelpBiz(event['venue_name'], locLat, locLng)
+    if yelpBiz:
+        researchVenue(yelpBiz)
+        eventObj = representation.eventRecord(yelpBiz.id, locLat, locLng, event['title'], event['start_time'], event['stop_time'], event['url'])
+        return eventObj
+
+# Fetching events from Google Calendar
+
+DAY_DATETIME = datetime.timedelta(days=1)
+scheduler = sched.scheduler(time.time, time.sleep)
```
Comment (jhugman) [inline app/request_handler.py]: Once we're starting to put things in schedulers, we are pretty far away from handling requests.

Suggest putting this into a separate file.

**Comment by jhugman** [comment]
I think the best we can do is admit this is tech debt, file an issue to fix it.

The correct course is to separate the firebase read/writing into its own file, then going from there.

**Comment by liuche** [comment]
[As a user, I want to be able to see events around me from the eventful service so I can use the app outside of Hawaii.](https://trello.com/c/7FZp1GZG/214-as-a-user-i-want-to-be-able-to-see-events-around-me-from-the-eventful-service-so-i-can-use-the-app-outside-of-hawaii)

**Review by jhugman — APPROVED** [review]

**Comment by liuche** [comment]
I'm assuming that this would merge after the island-crawler, so this updated pull request is rebased on top of that, and addresses the comments for both events pull requests.

**Comment by liuche** [comment]
Cherry-picked these onto Pull #56 .

**Pull request closed without merging**